ssl/statem/extensions_clnt.c

```diff
@@ -2015,3 +2015,119 @@ int tls_parse_stoc_psk(SSL_CONNECTION *s, PACKET *pkt,
 
     return 1;
 }
+
+EXT_RETURN tls_construct_ctos_client_cert_type(SSL_CONNECTION *sc, WPACKET *pkt,
+                                               unsigned int context,
+                                               X509 *x, size_t chainidx)
+{
+    sc->ext.client_cert_type_ctos = 0;
+#ifndef OPENSSL_NO_RPK
+    if ((sc->options & SSL_OP_RPK_CLIENT) != 0) {
+        if (!WPACKET_put_bytes_u16(pkt, TLSEXT_TYPE_client_cert_type)
+                || !WPACKET_start_sub_packet_u16(pkt)
+                || !WPACKET_start_sub_packet_u8(pkt)
+                || !WPACKET_put_bytes_u8(pkt, TLSEXT_cert_type_rpk)
+                || !WPACKET_put_bytes_u8(pkt, TLSEXT_cert_type_x509)
```

Comment:
This looks like it is not possible to disable X.509, and RPK is always additional. Perhaps OK for now, but it should be possible to offer just RPK, so that servers that understand the extension, but aren't willing to do RPK, and demand a client cert can bail out early.

Reply:
Yes, that was the point of my earlier comment. There might be a case for disabling certificates, and only go with RPK.